### PR TITLE
Color and similarity

### PR DESCRIPTION
**Pull request by DanieleMaselli**
Awesome you want to make a PR  
please see [https://github.com/fabianmoronzirfas/gestalten-in-code/blob/master/.github/CONTRIBUTING.md](https://github.com/fabianmoronzirfas/gestalten-in-code/blob/master/.github/CONTRIBUTING.md) for further infos.

### CODE REVIEW

**Comment by DanieleMaselli** [comment]
hope i got it right

File: color/p5js/Law of simmilarity + color/index.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---  
+layout: [p5js]
+author: [Daniele Maselli]
+author-url: https://github.com/DanieleMaselli
+title: color + similarity
```
Comment (fabianmoronzirfas) [inline color/p5js/Law of simmilarity + color/index.md]: Please give it a meaningful name. We can't have the `+` in the Title. Name it „5-dots“ or something more creative.

**Comment by DanieleMaselli** [comment]
insert the index.md file

File: color/p5js/Law of simmilarity + color/index.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+tags: [P5.js, color, "Law of "]
```
Comment (fabianmoronzirfas) [inline color/p5js/Law of simmilarity + color/index.md]: Law of what? please add in the Law you think that is right and don't put it into quotes.   

Wrong:    

    "Law of Similarity"  

Right:  

    Law of Similarity

File: color/p5js/Law of simmilarity + color/index.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+permalink: /color/color-and-similarity/
```
Comment (fabianmoronzirfas) [inline color/p5js/Law of simmilarity + color/index.md]: use the new title as permalink

File: color/p5js/Law of simmilarity + color/index.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+summary: creating two gestalt laws in one image.
```
Comment (fabianmoronzirfas) [inline color/p5js/Law of simmilarity + color/index.md]: This is a bit vague. Can write a bit more about what you are trying to do?

File: color/p5js/Law of simmilarity + color/index.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
+---  
+layout: [p5js]
```
Comment (fabianmoronzirfas) [inline color/p5js/Law of simmilarity + color/index.md]: No `[]` for the layout. See https://github.com/fabianmoronzirfas/gestalten-in-code/blob/master/.github/CONTRIBUTING.md#p5js

File: color/p5js/Law of simmilarity + color/index.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+author: [Daniele Maselli]
```
Comment (fabianmoronzirfas) [inline color/p5js/Law of simmilarity + color/index.md]: No `[]` for the author. See https://github.com/fabianmoronzirfas/gestalten-in-code/blob/master/.github/CONTRIBUTING.md#p5js

File: color/p5js/Law of simmilarity + color/index.md [hunk 7]
```diff
@@ -3,7 +3,7 @@ layout: [p5js]
 author: [Daniele Maselli]
 author-url: https://github.com/DanieleMaselli
 title: color + similarity
-tags: [P5.js, color, "Law of "]
+tags: [P5.js, color, "Law of color+similarity "]
```
Comment (fabianmoronzirfas) [inline color/p5js/Law of simmilarity + color/index.md]: You can't do that. Sorry. If you want you can add both laws. Like this:  

    tags: [P5.js, color, Law of Color, Law of Similarity]

File: color/p5js/Law of simmilarity + color/index.md [hunk 8]
```diff
@@ -3,7 +3,7 @@ layout: [p5js]
...
 permalink: /color/color-and-similarity/
```
Comment (fabianmoronzirfas) [inline color/p5js/Law of simmilarity + color/index.md]: Still not right. Please use a meaningful title.

File: color/p5js/Law of simmilarity + color/index.md [hunk 9]
```diff
@@ -3,7 +3,7 @@ layout: [p5js]
 author: [Daniele Maselli]
 author-url: https://github.com/DanieleMaselli
 title: color + similarity
```
Comment (fabianmoronzirfas) [inline color/p5js/Law of simmilarity + color/index.md]: Please add a meaningful title.

File: color/p5js/Law of simmilarity + color/index.md [hunk 10]
```diff
@@ -3,7 +3,7 @@ layout: [p5js]
...
-tags: [P5.js, color, "Law of "]
+tags: [P5.js, color, "Law of color+similarity "]
 permalink: /color/color-and-similarity/
 summary: creating two gestalt laws in one image.
```
Comment (fabianmoronzirfas) [inline color/p5js/Law of simmilarity + color/index.md]: Please wrote something informing.

File: color/p5js/Law of simmilarity + color/index.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
+function setup() {
+  
+  createCanvas (800,800)
```
Comment (fabianmoronzirfas) [inline color/p5js/Law of simmilarity + color/index.js]: The canvas needs the code following code:  

```js
  var canvas = createCanvas(100, 100); // !this is important
  canvas.parent('sketch'); // !this as well
```

as explained here https://github.com/fabianmoronzirfas/gestalten-in-code/blob/master/.github/CONTRIBUTING.md#p5js

Comment (fabianmoronzirfas) [inline color/p5js/Law of simmilarity + color/index.js]: The `;` is missing.

File: color/p5js/Law of simmilarity + color/index.js [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+  background (0);
+  
+
+}
+
+function draw() {
+  fill(65, 95, 98);
+  ellipse(200,200, 100, 100);
```
Comment (fabianmoronzirfas) [inline color/p5js/Law of simmilarity + color/index.js]: Please make it dynamic. Don't repeat yourself. These 4 Dots can be crated in a loop. Also the location should be dependent on the canvas size.

**Comment by fabianmoronzirfas** [comment]
Hi @DanieleMaselli thank you for the PR. Can you please be a bit more verbose about the description and less verbose in the title? I actually don't get why it is the "Law of Color"? We haven't yet expressed this law. So if you wan't to add it you should also write the definition for that.  

You index.js is still missing the `canvas.parent` thingy as described in the guide. Also the name for the folder should not be the "law of something something". Please take a look into the site https://interface.fh-potsdam.de/gestalten-in-code/ how it is structured and read the guide thoroughly.

**Comment by DanieleMaselli** [comment]
Hi @fabianmoronzirfas i will correct the PR ASAP..i just got some problem for rename my file. The law of color means that your mind is gonna focus on the colored dot, together with similarity for the other dots.

**Comment by fabianmoronzirfas** [comment]
> The law of color means that your mind is gonna focus on the colored dot
Well this is a bit thin. If you wan't to introduce a new law you should have some more then "you are going to focus on the colored dot" I can reverse this if I have one black dot between several colored ones I will focus on the black one.   

Also your index.js is still wrong.

File: color/p5js/Law of simmilarity + color/index.js [hunk 13]
```diff
@@ -1,11 +1,15 @@
+canvas.parent('sketch');
```
Comment (fabianmoronzirfas) [inline color/p5js/Law of simmilarity + color/index.js]: This has to be called after assigning the return value of `createCanvas` to the variable `canvas`

File: color/p5js/Law of simmilarity + color/index.js [hunk 14]
```diff
@@ -1,11 +1,15 @@
...
   background (0);
   
-
 }
 
+
+
 function draw() {
   fill(65, 95, 98);
   ellipse(200,200, 100, 100);
```
Comment (fabianmoronzirfas) [inline color/p5js/Law of simmilarity + color/index.js]: Still not dynamic from here on.

**Comment by fabianmoronzirfas** [comment]
seariously? "dots that get your attention" should be the title?

**Comment by DanieleMaselli** [comment]
renamed the folder! also the code is fixed...

File: color/p5js/colored-dots/index.js [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+	canvas = createCanvas(800, 800);
+	canvas.parent('sketch');	
+	background(0);
+    
+    // sizes
+	CORNER_WIDTH  = 100;
+	CORNER_HEIGHT = CORNER_WIDTH;
+
+	// positions
+	corners = [
```
Comment (fabianmoronzirfas) [inline color/p5js/colored-dots/index.js]: 😆  this is still static but nice try.

File: color/p5js/colored-dots/index.html [hunk 16]
```diff
@@ -0,0 +1,17 @@
+<html>
```
Comment (fabianmoronzirfas) [inline color/p5js/colored-dots/index.html]: please remove the index.html file we don't need it.

File: color/p5js/colored-dots/index.md [hunk 17]
```diff
@@ -0,0 +1,9 @@
+---  
+layout: p5js
+author: Daniele Maselli
+author-url: https://github.com/DanieleMaselli
+title: colored-dots
+tags: [P5.js, color, Law of Similarity]
+permalink: /color/similarity/
+summary: items that are similar tend to be grouped together.
```
Comment (fabianmoronzirfas) [inline color/p5js/colored-dots/index.md]: "Items" not i"tems"

File: color/p5js/colored-dots/index.js [hunk 18]
```diff
@@ -0,0 +1,43 @@
+var canvas,
+	CORNER_WIDTH,
+	CORNER_HEIGHT,
+	corners,
+	c_color,
+	cc_color;
+
+function setup() {
+
+	canvas = createCanvas(800, 800);
```
Comment (fabianmoronzirfas) [inline color/p5js/colored-dots/index.js]: Please make the canvas 500 × 500 px. As all the other examples are and as it is requested in the guide.

**Comment by fabianmoronzirfas** [comment]
@DanieleMaselli don't give up. We are nearly there! 👍

**Comment by DanieleMaselli** [comment]
Never!!! 💪 here we go

**Comment by fabianmoronzirfas** [comment]
👍  🚀  🎉  Thank you.

**Pull request merged**